aws/modules/cloudtrail/main.tf

```diff
@@ -30,10 +31,47 @@ resource "aws_s3_bucket" "cloudtrail_bucket" {
 	force_destroy = var.bucket_force_destroy
 	policy        = data.aws_iam_policy_document.cloudtrail_s3_policy.json
 
-	server_side_encryption_configuration {
-		rule {
-			apply_server_side_encryption_by_default {
-				sse_algorithm = var.bucket_sse_algorithm
+	versioning {
+		enabled = var.bucket_enable_versioning
+	}
+
+	dynamic "logging" {
+		for_each = var.bucket_enable_logs == true ? [1] : []
+		content {
+			target_bucket = aws_s3_bucket.cloudtrail_log_bucket[0].id
+			target_prefix = "log/"
+		}
+	}
+
+	dynamic "server_side_encryption_configuration" {
+		for_each = var.bucket_enable_encryption == true ? [1] : []
+		content {
+			rule {
+				apply_server_side_encryption_by_default {
+					sse_algorithm = var.bucket_sse_algorithm
+				}
+			}
+		}
+	}
+}
+
+resource "aws_s3_bucket" "cloudtrail_log_bucket" {
+	count         = var.use_existing_cloudtrail ? 0 : 1
+	bucket        = local.log_bucket_name
+	force_destroy = var.bucket_force_destroy
+	acl           = "log-delivery-write"
+
+	versioning {
+		enabled = var.bucket_enable_versioning
+	}
+
+	dynamic "server_side_encryption_configuration" {
+		for_each = var.bucket_enable_encryption == true ? [1] : []
```

Comment:
can you use `count` instead?

Reply:
To my knowledge, `count` cannot be used with a dynamic block.

Reply:
True! Thanks for the confirmation 💯